### PR TITLE
Sp 10 registration page

### PR DESCRIPTION
**Pull request by nenovKrasimir**

### CODE REVIEW

File: app/Http/Controllers/AuthController.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Support\Str;
+
+class AuthController extends Controller
+{
+
+    public function create_user(Request $request)
+    {
+        request()->validate([
+            'username'=>'required|unique:users',
```
Comment (realestateSoftuni) [inline app/Http/Controllers/AuthController.php]: I think is good idea to add |max:255|string|

File: app/Http/Controllers/AuthController.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        return response()->json(['message' => 'Your account has been registered. Please check your email to verify.'], 201);
+    }
+
+    public function verify($token)
+    {
+        $user = User::where('remember_token', '=', $token)->first();
+        if(!empty($user))
+        {
+            $user->email_verified_at = date('Y-m-d H:i:s');
+            $user->remember_token = Str::random(40);
```
Comment (realestateSoftuni) [inline app/Http/Controllers/AuthController.php:45]: maybe 20 characters is enough

File: app/Http/Controllers/AuthController.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            $user->is_confirmed = True;
+            $user->save();
+        }
+        else
+        {
+            abort(404);
```
Comment (realestateSoftuni) [inline app/Http/Controllers/AuthController.php]: We can log some message here

File: resources/js/components/EmailVerify.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import axios from 'axios';
+
+const VerifyEmail = () => {
+    const { token } = useParams();
+    const [loading, setLoading] = useState(true);
+    const [message, setMessage] = useState('');
+
```
Comment (realestateSoftuni) [inline resources/js/components/EmailVerify.jsx:12]: thanks for the comments is good practice

File: resources/js/components/NotFound.jsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// NotFound.jsx
+import React from 'react';
+
```
Comment (realestateSoftuni) [inline resources/js/components/NotFound.jsx:3]: We can use some cool template for this page , https://colorlib.com/wp/free-404-error-page-templates/   this site is only for example fell free to use other

File: resources/js/components/UserRegister.jsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    };
+
+    const onSubmit = async e => {
+        e.preventDefault();
+        setErrors({}); // Reset errors on new submission
+        setSuccess(''); // Reset success message on new submission
+        setIsLoading(true);
+
+        try {
+            const response = await axios.post('http://localhost:8000/api/create_user', formData);
```
Comment (realestateSoftuni) [inline resources/js/components/UserRegister.jsx]: localhost? I think we need to set variable and change it for the different environments

**Review by realestateSoftuni — CHANGES_REQUESTED** [review]

File: resources/js/components/EmailVerify.jsx (hunk 4 above)
Comment (nenovKrasimir) [reply resources/js/components/EmailVerify.jsx:12]: Done -> Added more comments.

File: app/Http/Controllers/AuthController.php (hunk 2 above)
Comment (nenovKrasimir) [reply app/Http/Controllers/AuthController.php:45]: Done

File: app/Http/Controllers/AuthController.php (hunk 1 above)
Comment (nenovKrasimir) [reply app/Http/Controllers/AuthController.php]: Done

File: app/Http/Controllers/AuthController.php (hunk 3 above)
Comment (nenovKrasimir) [reply app/Http/Controllers/AuthController.php]: Done

File: resources/js/components/UserRegister.jsx (hunk 6 above)
Comment (nenovKrasimir) [reply resources/js/components/UserRegister.jsx]: Done

**Comment by realestateSoftuni** [comment]
Now evriting look good thanks i will merge it after home page is ready

**Pull request closed without merging**